src/backend/joseki.be/webapp/Database/Cache/OwnershipCache.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using joseki.db;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+using webapp.Database.Models;
+
+namespace webapp.Database.Cache
+{
+    /// <summary>
+    /// Keeps track of Ownership items to reduce amount of interactions with real database.
+    /// </summary>
+    public class OwnershipCache : IOwnershipCache
+    {
+        private static readonly string CacheName = "ownership_cache";
+
+        private readonly JosekiDbContext db;
+        private readonly IMemoryCache cache;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OwnershipCache"/> class.
+        /// </summary>
+        /// <param name="db">Joseki database instance.</param>
+        /// <param name="cache">In-memory cache.</param>
+        public OwnershipCache(JosekiDbContext db, IMemoryCache cache)
+        {
+            this.db = db;
+            this.cache = cache;
+        }
+
+        /// <summary>
+        /// Returns the Ownership entries.
+        /// In this scenario it is updated daily, so the expiration is set to 1 day.
+        /// If the ownership will be updated manually, this cache should be invalidated.
+        /// </summary>
+        public async Task<List<OwnershipEntity>> GetEntries()
+        {
+            if (!this.cache.TryGetValue<IEnumerable<OwnershipEntity>>(CacheName, out var entries))
+            {
+                var db_entries = await this.db.Set<OwnershipEntity>().AsNoTracking().ToListAsync();
+                entries = this.cache.Set(CacheName, db_entries, absoluteExpiration: DateTimeOffset.Now.AddDays(1));
```

Comment:
What about storing in cache only <component_id, owner> tuple? Likely, It decreases used memory in a few times.

Reply:
Done, using _ValueTuples_ now.